tests/php/test_class-wc-connect-shipping-method.php

```diff
@@ -99,4 +99,64 @@ public function test_is_valid_package_destination( $package, $expected ) {
 
 	}
 
+	/**
+	 * @covers WC_Connect_Shipping_Method::calculate_shipping
+	 */
+	public function test_invalid_destination_calculate_shipping() {
+
+		$loader = new WC_Connect_Loader();
+		$loader->load_dependencies();
+
+		$shipping_method = $this->getMockBuilder( 'WC_Connect_Shipping_Method' )
+			->disableOriginalConstructor()
+			->setMethods( array( 'is_valid_package_destination', 'log' ) )
+			->getMock();
+
+		$shipping_method->expects( $this->any() )
+			->method( 'is_valid_package_destination' )
+			->will( $this->returnValue( false ) );
+
+		$shipping_method->expects( $this->once() )
+			->method( 'log' )
+			->with(
+				$this->stringContains( 'Package destination failed validation.' ),
+				$this->anything()
+			);
+
+		$shipping_method->calculate_shipping();
```

Comment:
In order to do so, we'd need to mock the API client as well, and so on. It's rather involved because we can't simply check the return value.. we have to make sure other methods weren't called.